src/js/api.js

```diff
@@ -24,17 +24,31 @@ class API {
    * @return {Promise}         The wrapped promise.
    */
   _wrap(promise) {
-    return promise.catch(function(res) {
-      // Only expose the error entity object, in the following form:
+    return promise.catch(res => {
+      if (!res.entity || res.error === "loaderror") {
+        throw new Error("Server seems to be down");
+      }
+      // res.entity is in the following form:
       // {"errno":104,
       //  "message":"Please authenticate yourself to use this endpoint.",
       //  "code":401,
       //  "error":"Unauthorized"}
-      throw res.entity;
+      throw Object.assign(new Error(), res.entity);
+    }).then(res => {
+      // Get rid of the .entity intermediate property.
+      return res.entity;
     });
-    //.done(function(res) {
-    //   return res.entity;
-    // })
+  }
+
+  createClient() {
+    return rest
+      .wrap(pathPrefix, {prefix: this.baseUrl})
+      .wrap(errorCode, {code: 400})
+      .wrap(defaultRequest, {headers: {"X-Requested-With": "readinglist-client"}})
```

Comment:
The  `X` header had been deprecated. See RFC 6648 — http://tools.ietf.org/rfc/rfc6648.txt